### PR TITLE
MDM-11453

### PR DESCRIPTION
**Pull request by axelhodler**

### CODE REVIEW

File: src/security/authenticated_user.js [hunk 1]
```diff
@@ -50,6 +50,9 @@ var AuthenticatedUser = mCAP.User.extend({
     mCAP.User.prototype.setReferencedCollections.apply(this, [attr]);
 
     if (attr.preferences && !(attr.preferences instanceof mCAP.Model) && this.get('preferences')) {
+      attr.preferences = _.omit(attr.preferences, function(value, key) {
+        return key.match('^custom') && value === '';
```
Comment (Zarlex) [inline src/security/authenticated_user.js]: Actually `.match()` expects a regex. Probably match will cast a string to regex internally. Anyway i would prefer `key.match(/^custom/)`

File: test/authentication/authentication.js [hunk 2]
```diff
@@ -39,6 +39,24 @@ describe("mCAP.authentication", function () {
     expect(user.hasRoleWithUuid('GROUP_NAME_ONE')).toBe(true);
     expect(user.hasRoleWithUuid('GROUP_NAME_TWO')).toBe(false);
   });
+
+  it('ignores custom attributes with empty strings', function() {
+    var user = new AuthenticatedUser({uuid: '09F40863-DD69-49AE-9F8C-C83EA3BCD6BA'});
+    user.set('preferences', new UserPreferences());
+
+    user.setReferencedCollections({
```
Comment (Zarlex) [inline test/authentication/authentication.js]: actually you would call
````
user.get('preferences').set({
  custom1: '',
  ...
});
```
Set referencedCollection should not be called. The method should only be called by `parse`and `set` to make sure that the nested attributes are not overwritten

File: test/authentication/authentication.js [hunk 3]
```diff
@@ -39,6 +39,24 @@ describe("mCAP.authentication", function () {
     expect(user.hasRoleWithUuid('GROUP_NAME_ONE')).toBe(true);
     expect(user.hasRoleWithUuid('GROUP_NAME_TWO')).toBe(false);
   });
+
+  it('ignores custom attributes with empty strings', function() {
+    var user = new AuthenticatedUser({uuid: '09F40863-DD69-49AE-9F8C-C83EA3BCD6BA'});
+    user.set('preferences', new UserPreferences());
```
Comment (Zarlex) [inline test/authentication/authentication.js]: `preferences` is a nested model so it is already set and you don't have to set it manually. Backbone sets all nested models/collections during initialisation

File: src/security/authenticated_user.js [hunk 4]
```diff
@@ -51,7 +51,7 @@ var AuthenticatedUser = mCAP.User.extend({
 
     if (attr.preferences && !(attr.preferences instanceof mCAP.Model) && this.get('preferences')) {
```
Comment (gitbreaker222) [inline src/security/authenticated_user.js]: I suggest a helper function `attr.preferences = removeEmptyStringValues(attr.preferences)`, but maybe this is too much

Comment (axelhodler) [inline src/security/authenticated_user.js]: no, would have been a good idea 👍

**Pull request merged**